src/plugins/explore/public/application/legacy/discover/application/components/visualizations/line/line_chart_rules.ts

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { VisualizationRule } from '../types';
+import { createLineConfig } from './line_vis_config';
+
+export const lineChartRule: VisualizationRule = {
+  name: 'lineChart',
+  matches: (numericalColumns, categoricalColumns, dateColumns) => {
+    return (
+      (numericalColumns.length === 1 && dateColumns.length === 1) || // Rule 1
+      (numericalColumns.length === 2 && dateColumns.length === 1) || // Rule 2
+      (numericalColumns.length === 1 &&
+        categoricalColumns.length === 1 &&
+        dateColumns.length === 1) || // Rule 3
+      (numericalColumns.length === 1 && categoricalColumns.length === 2 && dateColumns.length === 1) // Rule 4
+    );
+  },
+  createConfig: () => createLineConfig(),
+};
```

Comment:
1. Can we have unit tests for this (and other files) that have logic?

2. (Nitpick? Open for discussion) When i see all these `something.length === X || ...`, its hard to see what it is actually doing. Would it make sense to encapsulate these different rules in a variable with a meaninful name?